### DISCUSSION TITLE
Need help with colors in CSS

### DISCUSSION THREAD

**Discussion by pablitoSeija** [community/community#111412] [Programming Help] [answered]
### Body

I can't understand all the values of the colors RGB RGBA HEX. can anyone give me a better example? It is so difficult. Thanks...

### Guidelines

- [x] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by shoyim** [comment]
RGB (Red, Green, Blue):
> rgb(255, 0, 0) represents pure red.
> rgb(0, 255, 0) represents pure green.
> rgb(0, 0, 255) represents pure blue.
> rgb(255, 255, 0) represents yellow (combination of red and green).
RGBA (Red, Green, Blue, Alpha):
HEX is another way to represent RGB values but in a hexadecimal format. It's commonly used in web development.
Each channel (red, green, blue) is represented by a two-digit hexadecimal number, ranging from 00 to FF.
The format is #RRGGBB (where RR is the red value, GG is the green value, and BB is the blue value).
Example:
#FF0000 represents pure red.
#00FF00 represents pure green.
#0000FF represents pure blue.
#FFFF00 represents yellow.

RGB and RGBA use decimal values for color channels (0 to 255 for RGB and 0.0 to 1.0 for RGBA).
HEX uses hexadecimal values (00 to FF) for each color channel.

**Comment by samuelavike1** [comment]
RGB (Red, Green, Blue)
These are the three primary colors of light. By adjusting the intensity of each color (from 0, completely off, to 255, full brightness), you can create a vast range of colors.

Think of separate dimmers for red, green, and blue lights. Turning a dimmer up makes its color brighter.

Example:
Red: rgb(255, 0, 0) - This is full red because red is set to its maximum value (255) and green and blue are off (0).
Green: rgb(0, 255, 0) - Full green with red and blue off.
Yellow: rgb(255, 255, 0) - Mixing full red and green creates yellow.
White: rgb(255, 255, 255) - All three colors at full intensity produce white light.
Black: rgb(0, 0, 0) - All lights off results in black.
RGBA (Red, Green, Blue, Alpha): This extends RGB by adding an "alpha" channel (0 to 1) that controls an element's transparency.

Imagine placing a sheet in front of your colored lights. A higher alpha value (closer to 1) makes the color less transparent (more opaque), while a lower alpha (closer to 0) makes it more transparent.

Example:
rgba(255, 0, 0, 0.5) - This is a red color with 50% transparency (like a red filter sheet).
HEX (Hexadecimal): This is a shorthand way to represent an RGB color using a 6-digit hexadecimal code (combination of letters and numbers from 0 to F).
Think of it as a color code for your stage lights.

Example:
#FF0000 - This is the same red color as rgb(255, 0, 0), where FF represents full red, 00 represents no green or blue.
#FFFFFF - White (#FFF for each color).
#000000 - Black (#00 for each color).

Here are some online tools that can help you visualize and experiment with different color codes:
https://imagecolorpicker.com/
https://rgbacolorpicker.com/

**Comment by ByDiamondGod** [comment, marked as answer]
The reality is that they all have the same function, I recommend you to use the hexadecimal, it is the most used.

- HEXADECIMAL

#RRGGBB
Red green blue

Imagine three colored lights/lamps.

RED: red off (00) - red on (FF) 
GREEN: green off (00) green on (FF)
BLUE: blue off (00) blue on (FF) 

To add an opacity to the hexadecimal pattern HEX + A

No opacity #RRGGBB
With opacity #RRGGBBAA (AA = Opacity/Alpha).

Extra tip:

When the two values of a channel are the same, they can be abbreviated to become one: #33FF99 is the same color as #3F9 (#000 = black | #FFF = white).

---

If you want to know about HSL HSLA

Hue (hue - hue - hue - hue): Represents the color itself and is measured in degrees on the color wheel, ranging from 0 to 360. For example, 0 degrees is red, 120 degrees is green and 240 degrees is blue.

Saturation (saturation - contrast): Indicates the intensity and is expressed as a percentage, ranging from 0% (gray, no saturation) to 100% (saturated color).

Lightness: Represents the lightness of the color and is measured as a percentage, ranging from 0% (black) to 100% (white).

Alpha (opacity): Controls the transparency of the color and is specified as a value between 0 and 1, where 0 is completely transparent and 1 is completely opaque.

Example: hsla(300, 100%, 20%, 0.5);

Hue: 300 degrees (a hue that could be a deep blue).
Saturation: 100% (fully saturated color).
Luminosity: 20% (a little dark).
Opacity: 0.5 (semi-transparent).

---

And some fast RGBA, following the same.

These are values from 0 to 255

RGB 0, 0, 0 = Black
RGB 255, 255, 255 = White

RGBA(255, 0, 0, 0, 0.5);

255 in the red component, 0 in the green component and 0 in the blue component) and has an opacity of 50%, which means it would be semi-transparent.

I hope it helps you, this documentation that I put together, helped me a lot when I first started.

---------
--------- ESP
---------

La realidad es que todos tienen la misma función, te recomiendo que utilices el hexadecimal, es el más utilizado.

- HEXADECIMAL

#RRGGBB
Rojo verde azul

Imagina tres luces/lámparas de colores.

ROJO: rojo apagado (00) - rojo encendido (FF) 
VERDE: verde apagado (00) verde encendido (FF)
AZUL: azul apagado (00) azul encendido (FF) 

Para añadir una opacidad al patrón hexadecimal HEX + A

Sin opacidad #RRGGBB
Con opacidad #RRGGBBAA (AA = Opacidad/Alfa).

- Consejo extra:

Cuando los dos valores de un canal son iguales, pueden abreviarse para convertirse en uno: #33FF99 es el mismo color que #3F9 (#000 = negro | #FFF = blanco).

---

Si querés saber de HSL HSLA

Hue (Grado de color (hue - tono - matiz): Representa el color en sí mismo y se mide en grados en el círculo cromático, que van desde 0 a 360. Por ejemplo, 0 grados es rojo, 120 grados es verde y 240 grados es azul.

Saturation (saturación - contraste): Indica la intensidad y se expresa como un porcentaje, que va desde 0% (gris, sin saturación) hasta 100% (color saturado).

Lightness (luminosidad - luz al color): Representa la claridad del color y se mide en porcentaje, que va desde 0% (negro) hasta 100% (blanco).

Alpha (opacidad): Controla la transparencia del color y se especifica como un valor entre 0 y 1, donde 0 es completamente transparente y 1 es completamente opaco.

Ejemplo: hsla(300, 100%, 20%, 0.5);

Tono: 300 grados (un tono que podría ser un azul intenso).
Saturación: 100% (color completamente saturado).
Luminosidad: 20% (un poco oscuro).
Opacidad: 0.5 (semi-transparente).

---

Y algo rápido de RGBA, siguiendo lo mismo.

Son valores del 0 al 255

RGB 0, 0, 0 = Negro
RGB 255, 255, 255 = Blanco

RGBA(255, 0, 0, 0.5);

255 en el componente rojo, 0 en el verde y 0 en el azul) y tiene una opacidad del 50%, lo que significa que sería semitransparente.

Espero que te sirva, esta documentación que arme, me ayudó mucho a mí al principio cuando inicie

**Comment by Zain-Muhammad** [comment]
Certainly! Let's break down **RGB**, **RGBA**, and **HEX** color codes into simpler terms:

###  **RGB (Red, Green, Blue)**
 **What it is:** A way to mix red, green, and blue light to create colors.
 **How it works:** Each color (Red, Green, Blue) can have a value between 0 (no color) and 255 (full intensity). 
 **Example:** `rgb(255, 0, 0)` makes bright red because red is at its maximum intensity, with no green or blue.

### RGBA (Red, Green, Blue, Alpha)
- **What it adds:** The "Alpha" value controls the opacity of the color, from 0 (completely transparent) to 1 (fully opaque).
- **Example:** `rgba(255, 0, 0, 0.5)` creates a semi-transparent red, because the alpha value is 0.5, making it 50% see-through.

### HEX (Hexadecimal)
- **What it is:** A compact way to represent RGB values using hexadecimal numbers (0-9 and A-F).
- **How it works:** It starts with `#`, followed by pairs of hexadecimal numbers for red, green, and blue.
- **Example:** `#FF0000` is bright red, equivalent to `rgb(255, 0, 0)`. The `FF` represents 255 in red, with `00` for green and blue.

**In essence,** both RGB and HEX allow you to specify colors by mixing red, green, and blue. RGBA adds the ability to adjust transparency. These systems offer a flexible way to define colors for digital displays, making it easier to create a wide range of shades and tones.
